services/users.js

```diff
@@ -21,7 +21,14 @@ async function signup (firstName, lastName, email) {
 }
 
 async function login (secret) {
-  const userData = await UserModel.getUserBySecret(secret)
+  let userData
+  const trueUserSecret = await UserModel.getTrueUserSecretBySecret(secret)
+  if (trueUserSecret == null) {
+    userData = await UserModel.getUserBySecret(secret)
+  } else if (trueUserSecret != null) {
```

Comment:
[nit] `else if` is unnecessary, can just be `else`

Reply:
(If we change `UserModel.getTrueUserSecretBySecret` then this doesn't matter, ofc)